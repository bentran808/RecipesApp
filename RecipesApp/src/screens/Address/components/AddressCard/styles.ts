import { StyleSheet } from 'react-native';

const styles = StyleSheet.create({
  container: {
    paddingVertical: 10,
    paddingHorizontal: 20
  },
  titleWrapper: {
    flexDirection: 'row',
    alignItems: 'center'
  },
  titleIcon: {
    height: 20,
    width: 20,
    marginRight: 10,
    tintColor: 'gray'
  },
  addressText: {
    color: 'gray',
    paddingVertical: 10
  },
  buttonWrapper: {
    flexDirection: 'row',
    justifyContent: 'space-around'
  }
});

export default styles;
