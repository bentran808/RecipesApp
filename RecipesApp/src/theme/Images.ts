export const BackArrow = require('assets/images/icons/backArrow.png');
export const MenuIcon = require('assets/images/icons/menu.png');
export const HomeIcon = require('assets/images/icons/home.png');
export const CategoryIcon = require('assets/images/icons/category.png');
export const SearchIcon = require('assets/images/icons/search.png');
export const TimeIcon = require('assets/images/icons/time.png');
export const CloseIcon = require('assets/images/icons/close.png');
export const BasketIcon = require('assets/images/icons/basket.png');
export const CartIcon = require('assets/images/icons/cart.png');
