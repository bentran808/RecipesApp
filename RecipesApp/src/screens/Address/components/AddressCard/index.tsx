import AppStyles from 'AppStyles';
import Button from 'components/Button';
import React, { useCallback } from 'react';
import { Image, ImageSourcePropType, Text, View } from 'react-native';
import { AddressModel } from 'store/AddressStore';
import styles from './styles';

type Props = {
  item: AddressModel;
  source: ImageSourcePropType;
  onPressEdit: (item: AddressModel) => void;
  onPressDelete: (item: AddressModel) => void;
};

const AddressCard = ({ item, source, onPressEdit, onPressDelete }: Props) => {
  const handlePressEdit = useCallback(() => {
    onPressEdit(item);
  }, [item]);

  const handlePressDelete = useCallback(() => {
    onPressDelete(item);
  }, [item]);

  return (
    <View style={styles.container}>
      <View style={styles.titleWrapper}>
        <Image source={source} style={styles.titleIcon} />
        <Text style={AppStyles.textBold}>{item.type}</Text>
      </View>
      <Text style={styles.addressText}>{item.address}</Text>
      <View style={styles.buttonWrapper}>
        <Button
          testID="editBtn"
          type="outlined"
          color="red"
          borderColor="red"
          paddingVertical={5}
          title="EDIT"
          onPress={handlePressEdit}
          style={{
            borderRadius: 5
          }}
        />
        <Button
          testID="deleteBtn"
          type="outlined"
          color="red"
          borderColor="red"
          paddingVertical={5}
          paddingHorizontal={50}
          title="DELETE"
          onPress={handlePressDelete}
          style={{
            borderRadius: 5
          }}
        />
      </View>
    </View>
  );
};

export default React.memo(AddressCard);
