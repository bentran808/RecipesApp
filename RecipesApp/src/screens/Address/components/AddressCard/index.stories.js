import { action } from '@storybook/addon-actions';
import { storiesOf } from '@storybook/react-native';
import { address } from 'mocks';
import React from 'react';
import CenterView from 'storybook/stories/CenterView';
import { HomeIcon } from 'theme';
import AddressCard from '.';

storiesOf('Address Card', module)
  .addDecorator((getStory) => <CenterView>{getStory()}</CenterView>)
  .add('default', () => (
    <AddressCard
      item={address[0]}
      source={HomeIcon}
      onPressEdit={action('clicked-edit')}
      onPressDelete={action('clicked-delete')}
    />
  ));
