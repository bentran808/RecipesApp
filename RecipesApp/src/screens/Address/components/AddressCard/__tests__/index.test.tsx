import { fireEvent, render } from '@testing-library/react-native';
import { address } from 'mocks';
import renderer from 'react-test-renderer';
import { HomeIcon } from 'theme';
import AddressCard from '..';

describe('Address Card Component', () => {
  const props = {
    item: address[0],
    source: HomeIcon,
    onPressEdit: jest.fn(),
    onPressDelete: jest.fn()
  };

  test('should render correctly', () => {
    const tree = renderer.create(<AddressCard {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('should call function handlePressEdit', () => {
    const { getByTestId } = render(<AddressCard {...props} />);
    const button = getByTestId('editBtn');

    fireEvent.press(button);

    expect(props.onPressEdit).toHaveBeenCalled();
  });

  test('should call function handlePressDelete', () => {
    const { getByTestId } = render(<AddressCard {...props} />);
    const button = getByTestId('deleteBtn');

    fireEvent.press(button);

    expect(props.onPressDelete).toHaveBeenCalled();
  });
});
