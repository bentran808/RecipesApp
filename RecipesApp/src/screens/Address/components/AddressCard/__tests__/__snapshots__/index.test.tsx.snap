// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Address Card Component should render correctly 1`] = `
<View
  style={
    Object {
      "paddingHorizontal": 20,
      "paddingVertical": 10,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
      }
    }
  >
    <Image
      source={1}
      style={
        Object {
          "height": 20,
          "marginRight": 10,
          "tintColor": "gray",
          "width": 20,
        }
      }
    />
    <Text
      style={
        Object {
          "fontWeight": "bold",
        }
      }
    >
      Home
    </Text>
  </View>
  <Text
    style={
      Object {
        "color": "gray",
        "paddingVertical": 10,
      }
    }
  >
    448 TNV
  </Text>
  <View
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "space-around",
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID="editBtn"
    >
      <View
        style={
          Array [
            Object {
              "borderRadius": 100,
              "marginHorizontal": 10,
            },
            Object {
              "borderColor": "red",
              "borderWidth": 1,
              "paddingHorizontal": 60,
              "paddingVertical": 5,
            },
            Object {
              "borderRadius": 5,
            },
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "red",
                "fontSize": 14,
                "fontWeight": "normal",
                "textAlign": "center",
              },
            ]
          }
        >
          EDIT
        </Text>
      </View>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID="deleteBtn"
    >
      <View
        style={
          Array [
            Object {
              "borderRadius": 100,
              "marginHorizontal": 10,
            },
            Object {
              "borderColor": "red",
              "borderWidth": 1,
              "paddingHorizontal": 50,
              "paddingVertical": 5,
            },
            Object {
              "borderRadius": 5,
            },
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "red",
                "fontSize": 14,
                "fontWeight": "normal",
                "textAlign": "center",
              },
            ]
          }
        >
          DELETE
        </Text>
      </View>
    </View>
  </View>
</View>
`;
